sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -32,138 +32,60 @@
-                if multi_record:
-                    logger.info("Request URL: %r", redacted_url)
-                    logger.info("Request method: %r", http_request.method)
-                    logger.info("Request headers:")
-                    for header, value in http_request.headers.items():
-                        logger.info("    %r: %r", header, value)
-                    if isinstance(http_request.body, types.GeneratorType):
-                        logger.info("File upload")
-                        return
-                    try:
-                        if isinstance(http_request.body, types.AsyncGeneratorType):
-                            logger.info("File upload")
-                            return
-                    except AttributeError:
-                        pass
-                    if http_request.body:
-                        logger.info("A body is sent with the request")
-                        return
-                    logger.info("No body was attached to the request")
-                    return
-                log_string = "Request URL: '{}'".format(redacted_url)
-                log_string += "\nRequest method: '{}'".format(http_request.method)
-                log_string += "\nRequest headers:"
-                for header, value in http_request.headers.items():
-                    log_string += "\n    '{}': '{}'".format(header, value)
-                if isinstance(http_request.body, types.GeneratorType):
-                    log_string += "\nFile upload"
-                    logger.info(log_string)
-                    return
-                try:
-                    if isinstance(http_request.body, types.AsyncGeneratorType):
-                        log_string += "\nFile upload"
-                        logger.info(log_string)
-                        return
-                except AttributeError:
-                    pass
-                if http_request.body:
-                    log_string += "\nA body is sent with the request"
-                    logger.info(log_string)
-                    return
-                log_string += "\nNo body was attached to the request"
-                logger.info(log_string)
-
-            except Exception as err:  # pylint: disable=broad-except
-                logger.warning("Failed to log request: %s", repr(err))
-        else:
-            super().on_request(request)
-
-
-    def on_response(self, request, response):  # pylint: disable=too-many-return-statements, too-many-statements
-        # type: (PipelineRequest, PipelineResponse) -> None
-        if self._enable_diagnostics_logging:
-            http_response = response.http_response
-            ir = http_response.internal_response
-            try:
-                logger = response.context["logger"]
-
-                if not logger.isEnabledFor(logging.INFO):
-                    return
-
+            if self._enable_diagnostics_logging:
+                ir = response.http_response.internal_response
```

Comment:
As mentioned, we can't really make use of the `internal_response` object, as we have no idea what it will be.

Reply:
was this because of the transport pipeline being customizable by users? do we have any examples in other SDKs on how one might go about doing that? trying to see if using other transports with our SDK would cause other issues